src/actions/registration/index.ts

```diff
@@ -98,7 +98,9 @@ export const createIdentity = (encodedEntropy: string): ThunkAction => async (
 
   dispatch(setDid(identityWallet.identity.did))
   dispatch(setLoadingMsg(loading.loadingStages[3]))
-  await backendMiddleware.setIdentityWallet(userVault, password)
+
+  await storageLib.store.didDoc(identityWallet.didDocument)
+  backendMiddleware.identityWallet = identityWallet
```

Comment:
I was a bit late with posting my review :D

But yeah I really don't think this is a good idea to set things on the backendMiddleware from outside!

The previous method just had a bit of a bad name, but I think it was useful to have it for encapsulation's sake even if it's not going to be used in many places

Reply:
👍 I was debating this a lot internally. Will change to getter / accessor.

Reply:
Actually I was arguing for more than a setter, basically a `BackendMiddleware.createIdentityWallet` method which takes the necessary steps depending on what it was provided. I'm not entirely sure about what its responsibilities would be though, been trying to refactor/move things around for a while yesterday and this morning, and it's still unclear tbh..

A note about here (`createIdentityWallet`) specifically though, this change would mean that `registry.authenticate` is not called after (presumably) successful identity creation. Do we want that?

Reply:
off topic and can be handled in a separate PR